ui/src/app/lib/atlasmap-data-mapper/services/initialization.service.ts

```diff
@@ -489,9 +489,10 @@ isSource=${docdef.initModel.isSource}, inspection=${docdef.initModel.inspectionT
       }
 
       this.cfg.mappingService.fetchMappings(mappingFiles, this.cfg.mappings).toPromise()
-        .then((result: boolean) => {
+        .then( async(result: boolean) => {
         this.cfg.initCfg.mappingInitialized = true;
         this.updateStatus();
+        await this.cfg.mappingService.notifyMappingUpdated();
```

Comment:
Can it be `...notifyMappingUpdated().then(() => resolve(true);)` instead? async-await tend to make more things async...